testutil/mocking_test.go

```diff
@@ -0,0 +1,62 @@
+package testutil_test
+
+import (
+	"fmt"
+
+	"github.com/snapcore/snapd/testutil"
+)
+
+func ExampleBackupBeforeMocking_mockingSimple() {
```

Comment:
Is there a reason why these can't be actual tests that check that the mocked values are restored?

Reply:
They are actual tests, and they do check these values. This is a technique of `go test` the purpose of which to provide documentation of [examples through testing](https://pkg.go.dev/testing#hdr-Examples).

You can check it, if you'd like, but just changing anything in the `// Output: ... ` comment.